src/pocketmine/Player.php

```diff
@@ -1768,7 +1768,7 @@ public function handleDataPacket(DataPacket $packet){
 
 				if($this->isCreative()){
 					$this->inventory->setHeldItemSlot(0);
-				}else{
+				}elseif(!$this->inventory->getHotbarSlotIndex(0) === null){
```

Comment:
By the way, if you wonder why i do so many pull requests.. I don't want to build the phar myself, because jenkins adds inprovements